<?xml version="1.0"?>
<launch>
    <!--set you robot's moveit pkg here-->
    <arg name = "moveit_pkg" default="ur5_moveit_config"/>
    <arg name="num_ik_threads" default="4"/>
    <arg name="num_ik_attempts" default="5"/>
    <arg name="use_service_ik" default="false"/>
    <arg name="use_flood_fill" default="false"/>
    <!--Launch moveit required pkgs for your robot-->

    <include file="$(find map_generation)/launch/moveit_pkgs.launch">
        <arg name="moveit_pkg" value="$(arg moveit_pkg)"/>
    </include>

    <!--Node for map generation-->
    <node name="map_gen" pkg="map_generation" type="map_generation_node" output="screen">
      <param name="group_name" value="manipulator" />
      <param name="ee_frame" value="tool0" />
      <param name="pkg_name" value="$(arg moveit_pkg)"/>
      <param name="resolution" value="0.5" />
      <param name="radius" value="1" />
      <param name="num_ik_threads" value="$(arg num_ik_threads)"/>
      <param name="num_ik_attempts" value="$(arg num_ik_attempts)"/>
      <param name="use_service_ik" value="$(arg use_service_ik)"/>
      <param name="use_flood_fill" value="$(arg use_flood_fill)"/>
      <param name="check_collision" value="true"/>
      <param name="path" value="$(find map_generation)/maps/" />
      <param name="filename" value="default" />

    </node>
</launch>
